libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1128 @@
+#include "AP_Mount_Tusuav.h"
+#if HAL_MOUNT_TUSUAV_ENABLED
+#include <AP_Param/AP_Param.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_RTC/AP_RTC.h>
+#include <GCS_MAVLink/GCS.h>
+#include <GCS_MAVLink/include/mavlink/v2.0/checksum.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define AP_MOUNT_TUSUAV_HEADER1       0x5A     // first header byte
+#define AP_MOUNT_TUSUAV_PACKETLEN_MIN 6        // min number of bytes in a packet (1 header bytes,1 flag,1 cmd,1 sub-cmd,length,1 crc)
+#define AP_MOUNT_TUSUAV_DATALEN_MAX   (AP_MOUNT_TUSUAV_PACKETLEN_MAX-AP_MOUNT_TUSUAV_PACKETLEN_MIN) // max bytes for data portion of packet
+#define AP_MOUNT_TUSUAV_HEALTH_TIMEOUT_MS 1000 // state will become unhealthy if no attitude is received within this timeout
+#define AP_MOUNT_TUSUAV_SEND_INTERVAL_MS 100 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_PARSE_INTERVAL_MS 1 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_TIME_SYNC_INTERVAL_MS 1000 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_ZOOM_SPEED     500    // hard-coded zoom speed (fast)
+#define AP_MOUNT_TUSUAV_ZOOM_MAX       30      // hard-coded absolute zoom times max
+#define AP_MOUNT_TUSUAV_DEBUG 0
+#define debug(fmt, args ...) do { if (AP_MOUNT_TUSUAV_DEBUG) { GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Tusuav: " fmt, ## args); } } while (0)
+
+const char* AP_Mount_Tusuav::send_text_prefix = "Tusuav:";
+
+// init - performs any required initialisation for this instance
+// void AP_Mount_Tusuav::init()
```

Comment:
Thanks for the PR!  We normally don't allow commented out code to be added so could you remove this?

Reply:
Thanks for the feedback!  We've removed the commented-out code as requested.  Let us know if there’s anything else that needs adjustment